src/main/integrations/onuncaughtexception.ts

```diff
@@ -0,0 +1,57 @@
+import { getDefaultHub, Handlers, NodeClient } from '@sentry/node';
+import { Integration, SentryEvent, Severity } from '@sentry/types';
+import {
+  dialog,
+  // tslint:disable-next-line:no-implicit-dependencies
+} from 'electron';
+
+/** Capture unhandled erros. */
+export class OnUncaughtException implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = 'OnUncaughtException';
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(
+    private readonly options: {
+      onFatalError?(error: Error): void;
+    } = {},
+  ) {}
+
+  /**
+   * @inheritDoc
+   */
+  public install(): void {
+    global.process.on('uncaughtException', (error: Error) => {
+      getDefaultHub().withScope(async () => {
+        getDefaultHub().addEventProcessor(async (event: SentryEvent) => ({
+          ...event,
+          level: Severity.Fatal,
+        }));
+
+        const nodeClient = getDefaultHub().getClient() as NodeClient;
+        await nodeClient.captureException(error, getDefaultHub().getScope());
+
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
I'm not sure about this logic. When an app is packaged and run on a users machine, `NODE_ENV` will usually be undefined.

In the main process, you can detect if the app is packaged via:
```
process.defaultApp === undefined
```

Many will also want a way to not display the dialog in development as its 
a) too narrow to properly display stack traces
b) you can't copy and paste from the dialog